Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/pull_request.py

```diff
@@ -251,7 +251,7 @@ def pull_request_branch_point(cls, repository, args, **kwargs):
             source_remote = repository.source_remotes()[-1]
             args.remote = source_remote
 
-        if not repository.dev_branches.match(repository.branch) and (repository.branch is None or repository.branch in repository.DEFAULT_BRANCHES or
+        if not repository.dev_branches.match(repository.branch or '') and (repository.branch is None or repository.branch in repository.DEFAULT_BRANCHES or
```

Comment:
Can we not just totally avoid the match?

That said, I feel like the whole logic here is kinda over-complicated at this point, which makes it hard to understand.

Something like:

```diff
-        if not repository.dev_branches.match(repository.branch) and (
-            repository.branch is None
-            or repository.branch in repository.DEFAULT_BRANCHES
-            or repository.PROD_BRANCHES.match(repository.branch)
-            or repository.branch in repository.branches_for(remote=source_remote)
+        if repository.branch is None or (
+            not repository.dev_branches.match(repository.branch)
+            and (
+                repository.branch in repository.DEFAULT_BRANCHES
+                or repository.PROD_BRANCHES.match(repository.branch)
+                or repository.branch in repository.branches_for(remote=source_remote)
+            )
```

Realistically, I'd _hope_ we never had something that matched both `repo.dev_branches` *and* `repo.DEFAULT_BRANCHES`, or that matched both `repo.dev_branches` *and* `repo.prod_branches`, which means the only case that is actually interesting is the final `or` branch as the only one that _might_ match `repo.dev_branches` I think? Though maybe that's a bad assumption to rely on…

And if that's true, then we just need:

```diff
-        if not repository.dev_branches.match(repository.branch) and (
+        if (
             repository.branch is None
             or repository.branch in repository.DEFAULT_BRANCHES
             or repository.PROD_BRANCHES.match(repository.branch)
-            or repository.branch in repository.branches_for(remote=source_remote)
+            or (
+                not repository.dev_branches.match(repository.branch)
+                and repository.branch in repository.branches_for(remote=source_remote)
+            )
```

Reply:
I think you're right in that the logic could be simpler. Per other discussion, I've also moved it into a function and written some unit tests.

Reply:
I think you're right in that the logic could be simpler. Per other discussion, I've also moved it into a function and written some unit tests.